LayoutTests/http/tests/workers/service/resources/basic-timeout.js

```diff
@@ -1,75 +1,79 @@
 async function test()
 {
-    var urlResults = new Array;
-    var statusResults = new Array;
-    var headerResults = new Array;
+    let frameLoadedCount = 0;
+    const expectedFrameLoadedCount = 3;
 
-    function finishThisTest()
+    function addTestFrame(url, onloadFunction)
     {
-        for (var i = 0; i < urlResults.length; ++i) {
-            log("URL " + i + " - " + urlResults[i]);
-            log("Status code " + i + " - " + statusResults[i]);
-            log("Source' header " + i + " - " + headerResults[i]);
-        }
-        finishSWTest();
+        let iframe = frame.contentDocument.createElement("iframe");
+        iframe.src = url;
+        let iframeTimeoutHandle = setInterval(() => {
+            if (iframe.contentDocument.URL != "" && iframe.contentDocument.URL != "about:blank") {
+                clearTimeout(iframeTimeoutHandle);
+                onloadFunction();
+            }
+        }, 100);
+        iframe.onload = () => {
+            clearTimeout(iframeTimeoutHandle);
+            onloadFunction();
+        };
+        frame.contentDocument.body.appendChild(iframe);
+        return iframe;
     }
-    
-    try {
-        var frame = await interceptedFrame("resources/basic-timeout-worker.js", "/workers/service/resources/");
-        var fetch = frame.contentWindow.fetch;
 
-        if (window.testRunner)
-            testRunner.setServiceWorkerFetchTimeout(0);
-        
-        // The following two fetches should time out immediately
-        fetch("timeout-fallback.html").then(function(response) {
-            urlResults[0] = response.url;
-            statusResults[0] = response.status;
-            headerResults[0] = response.headers.get("Source");       
-        }, function(error) {
-            log(error);
-        });
+    frameText = function(testFrame)
```

Comment:
Why frameText = function(testFrame) instead of function frameText(testFrame)?

Reply:
It needs to have global scope otherwise I couldn't use it inside shouldBeEqualToString().